ktor-client/ktor-client-darwin/darwin/test/DarwinEngineTest.kt

```diff
@@ -20,10 +19,36 @@ import kotlin.test.*
 
 class DarwinEngineTest {
 
-    val testCoroutineContext: CoroutineContext = Dispatchers.Default
+    @Test
+    fun testRequestInRunBlockingDispatchersDefault() = runBlocking(Dispatchers.Default) {
```

Comment:
Why is Dispatchers.Default needed here and why can't Ktor internally do the right thing, so it's safe to be called from any dispatcher/thread?

Reply:
What do you mean by why is it needed? This tests that request can be made on default dispatcher. Below are the tests for other dispatchers, and no dispatcher.

Reply:
Sorry, I didn't see the one without dispatcher. But in our unit tests everything just blocks. Only Default doesn't block.

Reply:
Which version are you using? This fixed will be released in 2.3.0. You can try [EAP version](https://ktor.io/eap/), if you want to check the fix.

Reply:
I first tried with a local build, but with 2.3.0-eap-631 the issue is indeed fixed and apparently my repo had an outdated remote for the `main` branch. 🤦  Sorry for the confusion. Everything works as expected.

Reply:
Great, thanks for confirming.